Queries/Bonus.sql

```diff
@@ -0,0 +1,45 @@
+WITH BorrowingTimesInEachDay
+AS (
+	SELECT
+	   YEAR(DateBorrowed) AS yearNum,
+	   DATEPART(WEEK, DateBorrowed) AS weekNum,
+	   DATENAME(WEEKDAY, DateBorrowed) AS dayName,
+	   count(*) AS BorrowingTimesInTheWeek,
+	   sum(count(*)) OVER (PARTITION BY YEAR(DateBorrowed),DATEPART(WEEK, DateBorrowed))
+	   AS TotalLoansInTheWeek
+	FROM 
+	   Loans
+	GROUP BY
+	   YEAR(DateBorrowed),DATEPART(WEEK, DateBorrowed),DATENAME(WEEKDAY, DateBorrowed)
+	),
+PercentageOfLoansForEachDay
+AS (
+	SELECT
+	   yearNum,
+	   weekNum,
+	   dayName,
+	   BorrowingTimesInTheWeek,
+	   cast(BorrowingTimesInTheWeek AS DECIMAL(10, 2)) / cast(TotalLoansInTheWeek AS DECIMAL(10, 2)) * 100 
```

Comment:
what if TotalLoansInTheWeek is 0?

Reply:
there will not be any such case because the ``` BorrowingTimesInEachDay ``` deal with the ``` DateBorrowed ``` in the  ``` Loans``` table so there will not be any date there without borrowings.